front/src/applications/operationalStudies/hooks/useScenarioData.ts

```diff
@@ -178,14 +171,21 @@ const useScenarioData = (
     selectedTrainId,
     trainScheduleSummaries,
     trainSchedules,
-    trainScheduleUsedForProjection,
-    trainIdUsedForProjection,
-    projectedTrains,
-    allTrainsProjected,
+    projection:
+      trainScheduleUsedForProjection && projectionPath
+        ? {
+            trainSchedule: trainScheduleUsedForProjection,
+            path: projectionPath,
+            pathGeometry: projectionPathGeometry,
```

Comment:
If there is a projectionPath, we can be sure we will have a geometry no ?

```suggestion
            pathGeometry: projectionPathGeometry!,
```

Reply:
I modified the hook usePathProjection. Now it will either return `{ projectionPath: undefined, projectionPathGeometry: undefined }` or an object where `projectionPath` and `geometry` are defined.